src/common/TimeInterval.js

```diff
@@ -16,6 +16,9 @@ class TimeInterval {
 	toString() {
 		if (this.isBounded) {
 			// Using '_' as a separator character since it doesn't appear in ISO dates
+			if (this.startTimestamp === undefined || this.endTimestamp === undefined) {
```

Comment:
Does it even make sense to throw an error here? Should we be able to handle half-open time intervals.

Reply:
It absolutely makes sense to handle half-bounded intervals, but they should be considered unbounded. So a bounded time interval must have two endpoints, and if either of them is missing we have an error condition.
I don't think we need half-bounded intervals now (if we did, we'd have them) so this seems correct to me.

Reply:
Makes sense, I agree.